graphbolt/include/graphbolt/csc_sampling_graph.h

```diff
@@ -44,11 +45,19 @@ class CSCSamplingGraph : public torch::CustomClassHolder {
    * present.
    * @param type_per_edge A tensor representing the type of each edge, if
    * present.
+   * @param tensor_meta_shm The shared memory used to hold the meta information
+   * of the tensors associated with this class. Nullptr means that the graph is
+   * not on shared memory.
+   * @param tensor_data_shm The shared memory used to old the data of the
+   * tensors associated with this class. Nullptr means that the graph is not on
+   * shared memory.
    */
   CSCSamplingGraph(
       torch::Tensor& indptr, torch::Tensor& indices,
       const torch::optional<torch::Tensor>& node_type_offset,
-      const torch::optional<torch::Tensor>& type_per_edge);
+      const torch::optional<torch::Tensor>& type_per_edge,
+      SharedMemoryPtr&& tensor_meta_shm = nullptr,
+      SharedMemoryPtr&& tensor_data_shm = nullptr);
```

Comment:
I'd prefer the second solution, although it needs to define `BuildGraphFromSharedMemoryTensors` as a private static member function. The first solution may have overloading conflict since the only difference of the two constructors is from the default arguments.

Reply:
Make sense.